laws/src/main/scala/org/http4s/laws/discipline/ArbitraryInstances.scala

```diff
@@ -596,29 +596,27 @@ private[http4s] trait ArbitraryInstances {
     Cogen[(Byte, Byte, Byte, Byte)].contramap(ipv4 => (ipv4.a, ipv4.b, ipv4.c, ipv4.d))
 
   // https://tools.ietf.org/html/rfc3986#appendix-A
-  implicit val http4sTestingArbitraryForIPv6: Arbitrary[Uri.IPv6] = Arbitrary {
-    val h16 = timesBetween(min = 1, max = 4, genHexDigit.map(_.toString))
-    val ls32 = oneOf(h16 |+| const(":") |+| h16, getArbitrary[Uri.Ipv4Address].map(_.value))
-    val h16colon = h16 |+| const(":")
-    val :: = const("::")
-
-    oneOf(
-      times(6, h16colon) |+| ls32,
-      :: |+| times(5, h16colon) |+| ls32,
-      opt(h16) |+| :: |+| times(4, h16colon) |+| ls32,
-      opt(atMost(1, h16colon) |+| h16) |+| :: |+| times(3, h16colon) |+| ls32,
-      opt(atMost(2, h16colon) |+| h16) |+| :: |+| times(2, h16colon) |+| ls32,
-      opt(atMost(3, h16colon) |+| h16) |+| :: |+| opt(h16colon) |+| ls32,
-      opt(atMost(4, h16colon) |+| h16) |+| :: |+| ls32,
-      opt(atMost(5, h16colon) |+| h16) |+| :: |+| h16,
-      opt(atMost(6, h16colon) |+| h16) |+| ::
-    ).map(Uri.IPv6.apply)
+  implicit val http4sTestingArbitraryForIpv6Address: Arbitrary[Uri.Ipv6Address] = Arbitrary {
+    for {
+      a <- getArbitrary[Short]
```

Comment:
I chose not to take on the cats-scalacheck dependency to get mapN.  Since we publish the arbitraries, I thought it better to stay light.

Talk me out of it.

Reply:
These are equivalent, so I really don't care enough to argue.

It does the job right, but its not breaking. That being said. cats-scalacheck is one of those libraries that I think shouldn't be a microlibrary but instead in cats or at least recommended as these instances should be available everywhere.

Reply:
I was worried about the circular dependency and wary of relying on it, but I wasn't thinking clearly.

1. Putting it into scalacheck would be a circular dependency.  That would be bad.
2. Putting it into cats-laws would not add any new dependencies.
3. It can live on its own, with the only constraint that it not upgrade scalacheck higher than what's in cats-laws for the corresponding cats.  It's just a chore, not an impossibility, to make a compatible release.

We shouldn't be so scared of the status quo (3), but I see no reason not to implement (2).